src/commands/submit.rs

```diff
@@ -1,18 +1,23 @@
-use eyre::{Ok, Result};
+use color_eyre::Section;
+use eyre::{Ok, OptionExt, Result};
 
-use crate::{
-    git_client,
-    git_provider::{provider_factory, GitProvider},
-    project::settings::get_project_settings,
-};
+use crate::{git_client, git_provider::provider_factory, project::settings::get_project_settings};
 
 pub async fn submit() -> Result<()> {
-    let (provider, owner, repo) = git_client::get_git_client()?.get_repository_info()?;
-    let provider_obj = provider_factory(&provider)?;
+    let git_client = git_client::get_git_client()?;
+
+    let branch = git_client
+        .get_current_branch()
+        .ok_or_eyre("Failed to get the current branch")
+        .suggestion("Check whether you are checked out onto a branch")?;
+    let title = branch.clone();
```

Comment:
Do you need this clone id if you only immutably borrow title?

Reply:
I think not, but this is just a placeholder; in the future, the user will have the option to change the title of the PR